website/content/docs/concepts/stateful-deployments.mdx

```diff
@@ -0,0 +1,58 @@
+---
+layout: docs
+page_title: Stateful Deployments
+description: |-
+  Learn how Nomad handles Stateful Deployments.
+---
+
+# Stateful Deployments
+
+Nomad Stateful Deployments allow users to effectively bind their jobs to host
+[volumes]. On a lower level, this means taking Nomad's basic unit of workload
+placement—the [allocation]—and making it possible for users to have allocations
+that are ["sticky"] to volume IDs.
+
+<Note>
+Stateful Deployments support only Host Volumes. For CSI Volumes, we have <a
+href="/nomad/docs/job-specification/volume#per_alloc">per_alloc</a> property
+that serves a similar purpose.
+</Note>
+
+Stateful Deployments work on a per-task-group basis, that is, each task group
+can indicate that a host volume it requires must be "sticky." This is done by
+setting the parameter in the jobspec:
+
+```hcl
+job "app" {
+  group "example" {
+    # ...
+    volume "example" {
+      type      = "host"
+      source    = "ca-certificates"
+      read_only = true
+      sticky    = true
+    }
+    # ...
+  }
+}
+```
+
+If a volume is set to be sticky, during the deployment the scheduler adds its
+ID to the allocation that uses it. Stateful deployments require that any
+allocation that has the new VolumeID field set will only be (re)scheduled on
+nodes that have this Volume ID attached.
+
+Allocations that need to be rescheduled due to a destructive update will only
+ever be rescheduled on a node that has the host volume ID they used before.
+Scaling jobs with sticky volumes up results in more volume IDs claimed by the
+task group, but scaling jobs down does not delete unused volumes, nor does it
+touch the data present on them.
+
+If the scheduler cannot find a node that has the right volume ID present
+(perhaps because the node is down or disconnected), the allocation will be
+stopped and the job will remain in `pending` status until it's either stopped
+by the user or a feasible node is found.
```

Comment:
Saying "the allocation will be stopped" is a little confusing here... I think it's calling back to the previous paragraph where we're talking about rescheduled allocations, but the job would only be in a "pending" status for the first deployment, not reschedules. This paragraph applies for replacements or migrations as well.

Rather than calling out what happens to the old alloc or the job status, we should instead focus this paragraph on the resulting blocked eval. That's the primary symptom that a user will see, and it also includes the detail they need to know about why their alloc isn't getting placed.